wpilibc/src/main/native/cpp/simulation/DutyCycleEncoderSim.cpp

```diff
@@ -16,12 +16,49 @@ DutyCycleEncoderSim::DutyCycleEncoderSim(int channel) {
   frc::sim::SimDeviceSim deviceSim{"DutyCycle:DutyCycleEncoder", channel};
   m_simPosition = deviceSim.GetDouble("position");
   m_simDistancePerRotation = deviceSim.GetDouble("distance_per_rot");
+  m_simAbsolutePosition = deviceSim.GetDouble("absPosition");
+  m_simIsConnected = deviceSim.GetBoolean("connected");
+}
+
+double DutyCycleEncoderSim::Get() {
+  return m_simPosition.Get();
 }
 
 void DutyCycleEncoderSim::Set(units::turn_t turns) {
   m_simPosition.Set(turns.value());
+  m_simAbsolutePosition.Set(std::fmod(turns.value(), 1));
+}
+
+double DutyCycleEncoderSim::GetDistance() {
+  return m_simPosition.Get() * m_simDistancePerRotation.Get();
 }
 
 void DutyCycleEncoderSim::SetDistance(double distance) {
-  m_simPosition.Set(distance / m_simDistancePerRotation.Get());
+  double pos = distance / m_simDistancePerRotation.Get();
+  m_simPosition.Set(pos);
+  m_simAbsolutePosition.Set(std::fmod(pos, 1));
+}
+
+double DutyCycleEncoderSim::GetAbsolutePosition() {
+  return m_simAbsolutePosition.Get();
+}
+
+void DutyCycleEncoderSim::SetAbsolutePosition(double position) {
+  m_simAbsolutePosition.Set(position);
+}
+
+double DutyCycleEncoderSim::GetDistancePerRotation() {
+  return m_simDistancePerRotation.Get();
+}
+
+void DutyCycleEncoderSim::SetDistancePerRotation(double distancePerRotation) {
```

Comment:
Distance per rotation is designed to be read-only from the sim side (it's never read from in the main class).  So we shouldn't add a setDistancePerRotation accessor.